theano/tensor/nnet/abstract_conv.py

```diff
@@ -1699,11 +1761,19 @@ def conv(self, img, kern, mode="valid", dilation=1, num_groups=1):
                     for g in xrange(self.num_groups):
                         for n in xrange(output_channel_offset):
                             for im0 in xrange(input_channel_offset):
-                                # some cast generates a warning here
-                                out[b, g * output_channel_offset + n, ...] += _convolve2d(img[b, g * input_channel_offset + im0, ...],
-                                                                                          dilated_kern[g * output_channel_offset + n,
-                                                                                          im0, ...], 1, val, bval, 0)
+                                if unshared:
+                                    out[b, g * output_channel_offset + n, ...] += self.unshared2d(img[b, g * input_channel_offset + im0, ...],
+                                                                                                  dilated_kern[g * output_channel_offset + n, im0, ...],
+                                                                                                  out_shape[2:], direction)
+                                else:
+                                    # some cast generates a warning here
+                                    out[b, g * output_channel_offset + n, ...] += _convolve2d(img[b, g * input_channel_offset + im0, ...],
+                                                                                              dilated_kern[g * output_channel_offset + n, im0, ...],
+                                                                                              1, val, bval, 0)
+
         elif self.convdim == 3:
+            if unshared:
+                raise NotImplementedError('Unshared 3D convolution is not implemented')
```

Comment:
This should be asserted in the make_node to give an error when we build the graph instead of just when we execute the graph.